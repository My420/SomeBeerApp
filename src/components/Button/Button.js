import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const Button = ({
  children,
  className,
  disabled,
  activeStyle,
  onClick,
  ...attr
}) => {
  const buttonClassName = `${className} ${activeStyle}`;
  return (
    <button
      type="button"
      className={buttonClassName}
      disabled={disabled}
      onClick={onClick}
      {...attr}
    >
      {' '}
      {children}{' '}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  activeStyle: PropTypes.string
};

Button.defaultProps = {
  children: 'button',
  className: 'defaultButton',
  disabled: false,
  onClick: () => {},
  activeStyle: ''
};

export default Button;
