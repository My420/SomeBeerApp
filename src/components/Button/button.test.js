import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('test <Button /> component', () => {
  describe('test <Button /> component with no transferred props', () => {
    const wrapper = shallow(<Button />);

    test('should render default button', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test("should contain text 'button' by default", () => {
      expect(
        wrapper
          .find('button')
          .text()
          .trim()
      ).toBe('button');
    });

    test("should have class 'defaultButton'", () => {
      expect(wrapper.find('button').hasClass('defaultButton')).toBeTruthy();
    });

    test("attribute disabled should be equal 'false'", () => {
      expect(wrapper.find('button').prop('disabled')).toBeFalsy();
    });
  });

  describe('test <Button /> component with common props', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(
      <Button className="test" onClick={mockFunc}>
        test button
      </Button>
    );

    test('should render button with common attr', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should correct display button text', () => {
      expect(
        wrapper
          .find('button')
          .text()
          .trim()
      ).toBe('test button');
    });

    test('should have correct class attr', () => {
      expect(wrapper.find('button').hasClass('test')).toBeTruthy();
    });

    test('should call correct onClick function ', () => {
      wrapper.find('button').simulate('click');
      expect(mockFunc.mock.calls.length).toBe(1);
    });
  });

  describe('test <Button /> component with uncommon props', () => {
    const wrapper = shallow(
      <Button className="test" disabled activeStyle="active" type="submit" />
    );

    test('should render button  with uncommon attr', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should add activeClass props in class', () => {
      expect(wrapper.find('button').hasClass('test active')).toBeTruthy();
    });

    test("attribute disabled should be equal 'true'", () => {
      expect(wrapper.find('button').prop('disabled')).toBeTruthy();
    });

    test("button should have type 'submit'", () => {
      expect(wrapper.find('button').prop('type')).toBe('submit');
    });
  });
});
