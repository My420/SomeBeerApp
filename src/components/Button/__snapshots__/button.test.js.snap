// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Button /> component test <Button /> component with common props should render button with common attr 1`] = `
<button
  className="test "
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
   
  test button
   
</button>
`;

exports[`test <Button /> component test <Button /> component with no transferred props should render default button 1`] = `
<button
  className="defaultButton "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  button
   
</button>
`;

exports[`test <Button /> component test <Button /> component with uncommon props should render button  with uncommon attr 1`] = `
<button
  className="test active"
  disabled={true}
  onClick={[Function]}
  type="submit"
>
   
  button
   
</button>
`;
